krate/src/androidTest/java/hu/autsoft/krate/OptionalTests.kt

```diff
@@ -103,4 +104,18 @@ class OptionalTests {
         assertEquals(null, testKrate.optionalStringSet)
     }
 
+    @Test(expected = KrateValidationException::class)
+    fun testOptionalStringValidatedPrefFailsValidation() {
+        assertEquals(null, testKrate.optionalValidatedString)
+
+        testKrate.optionalValidatedString = "Lorem ipsum dolor sit amet"
+    }
+
+    @Test
+    fun testOptionalStringValidatedPassesValidation() {
```

Comment:
I like these two tests, and I'd also add a test for one of the non-optional types, for example a `Float`, just to cover a bit more ground.